hardware.inc

```diff
@@ -915,6 +916,19 @@ DEF BOOTUP_B_CGB    EQU %00000000
 DEF BOOTUP_B_AGB    EQU %00000001   ; GBA, GBA SP, Game Boy Player, or New GBA SP
 
 
+;***************************************************************************
+;*
+;* Interrupt vector addresses
+;*
+;***************************************************************************
+
+DEF INT_VBLANK EQU $0040
+DEF INT_STAT   EQU $0048
+DEF INT_TIMER  EQU $0050
+DEF INT_SERIAL EQU $0058
+DEF INT_JOYPAD EQU $0060

```

Comment:
These are too close to `IEF_*` constants... what about `INT_HANDLER_*`?

Reply:
That works, but is a little verbose in my opinion. Would `VEC_*` work, or does that make it unclear what they is for?

Reply:
Yes, that's why I considered those, but rejected them.

I also rejected `HANDLER_`, as I was afraid "handlers of *what*?" may arise; though considering this is a hardware companion file, don't we largely assume prior knowledge anyway?

Reply:
> These are too close to `IEF_*` constants... what about `INT_HANDLER_*`?

In what way are they too close to "IEF_"? Is it because they both start with the letter I and have three letters?

Reply:
I'm starting to have doubts about `INT_HANDLER_*` as well. From a hardware perspective, these addresses don't "handle" anything, it just jumps to them. The handling part is the softwares responsibility.

How about `INT_VECTOR_*`?

Reply:
> > These are too close to `IEF_*` constants... what about `INT_HANDLER_*`?
> 
> In what way are they too close to "IEF_"? Is it because they both start with the letter I and have three letters?

Yes, which I'm afraid makes them too easy to confuse in e.g. auto-completion or when glancing over code.



> I'm starting to have doubts about `INT_HANDLER_*` as well. From a hardware perspective, these addresses don't "handle" anything, it just jumps to them. The handling part is the softwares responsibility.
> 
> How about `INT_VECTOR_*`?

The addresses themselves don't handle anything, but OTOH none does; the hardware regs act when *writing* to them, and these are only special when the CPU *jumps* to them under special circumstances. The naming would therefore mirror what they are meant to be used for.

I'm not fond of "vector" because, AIUI, "vector" implies that an address is somehow stored there and dereferenced (like the 65xx does).

Reply:
> I'm not fond of "vector" because, AIUI, "vector" implies that an address is somehow stored there and dereferenced (like the 65xx does).

That makes sense. I will rename them to `INT_HANDLER_*`.

Reply:
Just for completeness' sake: I believe their only common use will be

```z80
SECTION "VBlank handler", ROM0[INT_HANDLER_VBLANK]
```

where I don't think the name length is an issue.